### PR TITLE
Add tide/merge-method-squash option for automated squashing

### PR DESCRIPTION
**Pull request by fsmunoz**
This is related to the blog post proposed at https://github.com/kubernetes/contributor-site/pull/313 and the Prow/Tilde squashing method. This change adds a reference to the method to the guide.

### CODE REVIEW

**Comment by MadhavJivrajani** [comment]
This is documented here: https://github.com/kubernetes/community/tree/master/contributors/guide/contributor-cheatsheet#squashing-commits

Any particular reason to add it to this doc as well?

**Comment by fsmunoz** [comment]
Hi @MadhavJivrajani . To me it seems that a reference in the guide makes sense when discussing squashing, with the cheatsheet complementing it. I don't have, however, a strong opinion on it, so if the existing practice is to keep this information exclusively in the cheatsheet I would say that this PR isn't needed.

File: contributors/guide/github-workflow.md [hunk 1]
```diff
@@ -237,6 +237,8 @@ For mass automated fixups such as automated doc formatting, use one or more
 commits for the changes to tooling and a final commit to apply the fixup en
 masse. This makes reviews easier.
 
+An alternative to this manual squashing process is to use the Prow and Tide based automation that is configured in GitHub: adding a comment to your PR with `/label tide/merge-method-squash` will trigger the automation and the tooling will squash your commits. Using this approach simplifies things for those less familiar with git, but there are situations in which a manual operation will be more aligned with the desired outcomes; reviewers will have this in mind and can ask for manual squashing to be done.
```
Comment (sftim) [inline contributors/guide/github-workflow.md:240]:
```suggestion
An alternative to this manual squashing process is to use the Prow and Tide based automation that is configured in GitHub: adding a comment to your PR with `/label tide/merge-method-squash` will trigger the automation and the tooling will squash your commits. Using this approach simplifies things for those less familiar with git, but there are situations in where it's better to squash locally; reviewers will have this in mind and can ask for manual squashing to be done.

By squashing locally, you control the commit message(s) for your work, and can separate a large PR into logically separate changes.
For example: you have a pull request that is code complete and has 24 commits. You rebase this against the same merge base, simplifying the change to two commits. Each of those two commits represents a single logical change and the commit message summarizes what this is. Reviewers see that the set of changes are now understandable, and approve your PR.
```
?

Comment (fsmunoz) [reply contributors/guide/github-workflow.md:240]: Agreed.

**Comment by fsmunoz** [comment]
@sftim squashed the commits, was going to assign to parispittman but mayeb the `ok-to-test` should be done prior?

**Comment by palnabarun** [comment]
Instead of having context at multiple places, can we have an one-liner here with reference to the squashing commits guideline https://github.com/kubernetes/community/tree/master/contributors/guide/contributor-cheatsheet#squashing-commits ?

This way, we will maintain only one copy of this information. Please feel free to augment the contributor cheatsheet with any added information needed.

File: contributors/guide/github-workflow.md [hunk 2]
```diff
@@ -237,6 +237,11 @@ For mass automated fixups such as automated doc formatting, use one or more
 commits for the changes to tooling and a final commit to apply the fixup en
 masse. This makes reviews easier.
 
+An alternative to this manual squashing process is to use the Prow and Tide based automation that is configured in GitHub: adding a comment to your PR with `/label tide/merge-method-squash` will trigger the automation so that GitHub squash your commits onto the target branch once the PR is approved. Using this approach simplifies things for those less familiar with Git, but there are situations in where it's better to squash locally; reviewers will have this in mind and can ask for manual squashing to be done.
+
+By squashing locally, you control the commit message(s) for your work, and can separate a large PR into logically separate changes.
+For example: you have a pull request that is code complete and has 24 commits. You rebase this against the same merge base, simplifying the change to two commits. Each of those two commits represents a single logical change and each commit message summarizes what changes. Reviewers see that the set of changes are now understandable, and approve your PR.
```
Comment (palnabarun) [inline contributors/guide/github-workflow.md:243]: This is a very good explanation of why people should squash manually if they can. Thank you for adding this!

**Comment by sftim** [comment]
LGTM

**Comment by fsmunoz** [comment]
@palnabarun should we go ahead with this (and then adjust the cheatsheet separately), or drop this and replace it with a different one that targets exclusively the cheatsheet? As I said initially, I see a cheatsheet as something that includes a condensed version of things that are documented elsewhere, but I'll defer to more contextualised judgement on how it  has been used in k8s.

**Comment by fsmunoz** [comment]
This was recently published after some prolonged editing https://www.kubernetes.dev/blog/2022/12/12/prow-and-tide-for-kubernetes-contributors/, I think this change here still makes sense...

/remove-lifecycle stale

**Review by mrbobbytables — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**